graphql-datetime-autoconfigure-common/src/main/java/com/zhokhov/graphql/datetime/boot/GraphQLDateTimeAutoConfiguration.java

```diff
@@ -50,9 +51,10 @@ public GraphQLDate graphQLDate(GraphQLDateTimeProperties configurationProperties
     @ConditionalOnMissingBean
     public GraphQLLocalDate graphQLLocalDate(GraphQLDateTimeProperties configurationProperties) {
         final String name = configurationProperties.getLocalDate().getScalarName();
-        if (name == null) {
-            return new GraphQLLocalDate();
-        } else {
+        final String format = configurationProperties.getLocalDate().getFormat();
+        try {
+            return new GraphQLLocalDate(name, false, DateTimeFormatter.ofPattern(format));
+        } catch (IllegalArgumentException | NullPointerException e) {
```

Comment:
Under what condition there can be errors like `IllegalArgumentException` or `NullPointerException`?

Reply:
If the pattern is null -> NPE
If the patern is invalid (i.e. not a date pattern) -> IAE

Null would be the most frequent value, so try-catch might have a performance penalty as opposed to if-else statement. But we need the try-catch for IAE so I choose this option for readability. And since this code will run at system startup, the performance should not be a critical issue.

Reply:
I think not to need to catch this error. Just make a check, if not null use `new GraphQLLocalDate(name, false, DateTimeFormatter.ofPattern(format))`, if there is null, use `new GraphQLLocalDate(name, false)`, if the format if wrong just throws an error, not magically replace with a default value, it will be much more predicted behavior.